driver/conv_driver.hpp

```diff
@@ -2944,7 +2953,7 @@ int ConvDriver<Tgpu, Tref>::VerifyForward()
     auto tolerance = GetDefaultTolerance();
     // iGemm's deviation is higher than other algorithms.
     // The reason is most likely different order of computations.
-    if(is_fwd_igemm)
+    if(is_fwd_igemm || is_fwd_cellfft)
```

Comment:
Is it really so that cellfft needs bigger tolerance?

Reply:
According to our test of hundreds of cases, FWD and BWD can achieve the same accuracy as GEMM in most cases (1e-6), but in some cases it is 1e-5; wrw accounts for about half of each. Therefore, for convenience, the tolerance is set to 1e-5

Reply:
> ...the same accuracy as GEMM...

Hmm... Do you mean GEMM or Implicit GEMM (iGEMM)?

> ...but in some cases it is 1e-5...

Have you carefully checked that cellfft works properly in these cases, especially for Forward and Backward directions?

Reply:
> > ...the same accuracy as GEMM...
> 
> Hmm... Do you mean GEMM or Implicit GEMM (iGEMM)?
> 
> > ...but in some cases it is 1e-5...
> 
> Have you carefully checked that cellfft works properly in these cases, especially for Forward and Backward directions?

Yes, we checked, even in these cases, the accuracy of most elements is still <1e-6(the driver result display *e-8), the accuracy > 1e-5(driver display result *e-7) occurs when the tile of 32x32 is selected, because the larger the tile, the lower the accuracy (but the accuracy of 32x32 also meets the requirement of < 1e-6 in many cases) and i test with direct method of cpu in one seperate program also, this my compare accuracy code:

bool check( const float* a, const float* b, int n )
{
double s0=0;
double s1=0;
double ref=0;
for( int i=0; i<n; ++i ){
double ai=(double)a[i];
double bi=(double)b[i];
double d=ai-bi;
double dd=dd;
double aa=2.0ai*ai;
double x=dd/aa;
if(x>ref){ ref=x; }
s0+=dd;
s1+=aa;
}
return (sqrt(s0/s1)<1e-5);
}

Reply:
> > ...the same accuracy as GEMM...
> 
> Hmm... Do you mean GEMM or Implicit GEMM (iGEMM)?
> 
> > ...but in some cases it is 1e-5...
> 
> Have you carefully checked that cellfft works properly in these cases, especially for Forward and Backward directions?

Yes, Direct method or gemm method

Reply:
@XiuYuLi what error metric is computed by ```check()``` ? It is not a RMSE. Anyway the value returned by ```check()``` is highly influenced by the input data distribution.

Reply:
[off-topic] @XiuYuLi Please do not use editing others' comments for your replies. Basically nobody receive notifications about edits, so we didn't read your response and I saw it by accident.

Reply:
@Kirpich30000 FYI answer by @XiuYuLi:
> I'm not sure, but the 'check()' function uses a commonly used method for checking the accuracy and correctness of the results. In fact, we not only rely on the return value of the 'check()', also output it to a file for viewing. Of course, it doesn't matter. Whether it's in the miopen driver or in my own test program, it's all passed.As for the accuracy of the result, it is what it should be

Reply:
Reply to the [following comment](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/335#discussion_r474983224) from @XiuYuLi :
> I think the driver display '*e-x' and the tolerance is not same means. I can't confirm now because it's just for testing convenience, but the value of tolerance in test.cpp whether be changed or not , it has no effect on the calculation results

Reply:
@XiuYuLi Let me ask again the following question (from [this comment](https://github.com/ROCmSoftwarePlatform/MIOpen/pull/335#discussion_r474983224)):
- ___Does driver fail verification without these adjustments? On which configs?___

Reply:
This is not resolved.